Assets/Scripts/EnemyController.cs

```diff
@@ -37,8 +39,26 @@ void Start()
     void Update()
     {
         RangedAttack();
+
+        UpdateHash();
```

Comment:
I think it would be simpler to use the `OnBecameInvisible` and `OnBecameVisible` methods instead of updating `player.enemyHash` every frame.